packages/snaps-utils/src/structs.ts

```diff
@@ -461,3 +467,78 @@ export function createUnion<Type, Schema extends readonly Struct<any, any>[]>(
 ) {
   return validateUnion(value, struct, structKey, true);
 }
+
+// These types are copied from Superstruct, to mirror `assign`.
+export function mergeStructs<
+  ObjectA extends ObjectSchema,
+  ObjectB extends ObjectSchema,
+>(
+  A: Struct<ObjectType<ObjectA>, ObjectA>,
+  B: Struct<ObjectType<ObjectB>, ObjectB>,
+): Struct<ObjectType<Assign<ObjectA, ObjectB>>, Assign<ObjectA, ObjectB>>;
+export function mergeStructs<
+  ObjectA extends ObjectSchema,
+  ObjectB extends ObjectSchema,
+  ObjectC extends ObjectSchema,
+>(
+  A: Struct<ObjectType<ObjectA>, ObjectA>,
+  B: Struct<ObjectType<ObjectB>, ObjectB>,
+  C: Struct<ObjectType<ObjectC>, ObjectC>,
+): Struct<
+  ObjectType<Assign<Assign<ObjectA, ObjectB>, ObjectC>>,
+  Assign<Assign<ObjectA, ObjectB>, ObjectC>
+>;
+export function mergeStructs<
+  ObjectA extends ObjectSchema,
+  ObjectB extends ObjectSchema,
+  ObjectC extends ObjectSchema,
+  ObjectD extends ObjectSchema,
+>(
+  A: Struct<ObjectType<ObjectA>, ObjectA>,
+  B: Struct<ObjectType<ObjectB>, ObjectB>,
+  C: Struct<ObjectType<ObjectC>, ObjectC>,
+  D: Struct<ObjectType<ObjectD>, ObjectD>,
+): Struct<
+  ObjectType<Assign<Assign<Assign<ObjectA, ObjectB>, ObjectC>, ObjectD>>,
+  Assign<Assign<Assign<ObjectA, ObjectB>, ObjectC>, ObjectD>
+>;
+export function mergeStructs<
+  ObjectA extends ObjectSchema,
+  ObjectB extends ObjectSchema,
+  ObjectC extends ObjectSchema,
+  ObjectD extends ObjectSchema,
+  ObjectE extends ObjectSchema,
+>(
+  A: Struct<ObjectType<ObjectA>, ObjectA>,
+  B: Struct<ObjectType<ObjectB>, ObjectB>,
+  C: Struct<ObjectType<ObjectC>, ObjectC>,
+  D: Struct<ObjectType<ObjectD>, ObjectD>,
+  E: Struct<ObjectType<ObjectE>, ObjectE>,
+): Struct<
+  ObjectType<
+    Assign<Assign<Assign<Assign<ObjectA, ObjectB>, ObjectC>, ObjectD>, ObjectE>
+  >,
+  Assign<Assign<Assign<Assign<ObjectA, ObjectB>, ObjectC>, ObjectD>, ObjectE>
+>;
+
+/**
+ * Merge multiple structs into one, using superstruct `assign`.
+ *
+ * Differently from plain `assign`, this function also copies over refinements from each struct.
+ *
+ * @param structs - The `superstruct` structs to merge.
+ * @returns The merged struct.
+ */
+export function mergeStructs(...structs: Struct<any>[]): Struct<any> {
+  const mergedStruct = (assign as (...structs: Struct<any>[]) => Struct)(
```

Comment:
Why is this type assertion necessary?

Reply:
I'm assuming it is caused by the type declarations for Superstruct being overwritten. The type declarations does not contain types for the function with a rest parameter.

Without the cast we get `A spread argument must either have a tuple type or be passed to a rest parameter.`, but everything still works.

Reply:
To be clear, the underlying function does support a rest parameter: https://github.com/ianstormtaylor/superstruct/blob/88563ad3abb818ee4d129e2d4af3c75f5ef56462/src/structs/utilities.ts#L60